lib/nsq/logger.ex

```diff
@@ -0,0 +1,23 @@
+defmodule NSQ.Logger do
+  require Logger
+
+  defdelegate debug(message), to: Logger
+  defdelegate info(message), to: Logger
+  defdelegate error(message), to: Logger
+
+  def warn(message) do
+    case Version.compare(System.version(), "1.11.0") do
+      :lt -> Logger.warn(message)
+      _ -> Logger.warning(message)
+    end
+  end
```

Comment:
I think we could do this here just with warn: https://jeffkreeftmeijer.com/elixir-backwards-compatible-logger/
This removes the compile warning.
```suggestion
  case Version.compare(System.version(), "1.11.0") do
    :lt -> defdelegate warn(message), to: Logger
    _ -> defdelegate warn(message), to: Logger, as: :warning
  end
```

I think other than this bit the pr is fine.

Reply:
Ah I did follow that guide at first and still got a bunch of deprecation warnings 😢

Reply:
Latest test run looks okay to me :)

Reply:
Ok to merge, I am checking the other warnings.

Reply:
Hmm - actually I accidentally rebased over the commit with your suggestion - I actually get test failures once I try that locally though. I think I'm just going to keep things as-is right now since we are in a less noisy state.